os/arch/arm/src/armv8-m/up_assert.c

```diff
@@ -474,6 +478,18 @@ void up_assert(const uint8_t *filename, int lineno)
 
 	abort_mode = true;
 
+	uint32_t fault_location;
+
+        /* Extract the PC value of instruction which caused the abort/assert */
+
+	if (g_assert_location) {
```

Comment:
1. I will change `g_assert_location` to `system_exception_location` as suggested.
2. `caller_retaddr` can't be changed as the macro is defined with it.
https://github.com/Samsung/TizenRT/blob/316e807f7bd4b5345e42f1fb1a63c991e6981f16/os/include/tinyara/mm/mm.h#L244-L247
3. `asserted_location` for `fault_location`
    How about using `system_exception_location` here? 
    As it can contain both the assert failure location or the fault location, I think system_exception_location is better.
    I will change `g_assert_location` to `g_system_exception_location`.

Please let me know your thoughts.

Reply:
2. No. it's the same as `#define ARCH_GET_RET_ADDRESS(x) \`. It does not mean you should use `caller_retaddr`. It's the same as function argument. If function `A(int B, int C)` as prototype, you can call it as `A(F, G);`
3. What I mean from `system_exception_location` and `user_assert_location` is the location which system makes the exception and the location which user intends assert. But `fault_location` should cover both and it does not matter who (system or user) makes, it only concerns location where causes assertion. Whatever it is, it should include both cases, system makes exception and user intends assert.

Reply:
2. I will change caller_retaddr to `kernel_assert_location` as we use this only in the case of kernel assert failure
3. So as suggested we have `user_assert_location` and `system_exception_location`
    fault (user or kernel) -> fault_handler (here `system_exception_location` is set) -> up_assert
    user assert -> sv call (here `user_assert_location` is set) -> up_assert
    kernel assert -> up_assert (`kernel_assert_location` is set)
The final `fault_location` will be equal to `user_assert_location` or `kernel_assert_location` or `system_exception_location` .

Reply:
Others are ok to me, but I don't know fault_location. It could be considered as location from faults.

Reply:
Okay, I will use asserted_location for fault_location as suggested earlier.